x-pack/plugins/security_solution/docs/prebuilt_rules_customization_rfc.md

```diff
@@ -0,0 +1,2867 @@
+
+For prebuilt rules, the `rule_source.is_customized` value will be initially set to `false` when a brand new rule is installed, but will be rewritten to `true` if a rule's field is edited and diverges from the value from the base version of the rule.
+
+See section [Updating the `isCustomized` flag](#updating-the-is_customized-field)
+
+#### `source_updated_at` subfield
+
+The `source_updated_at` will be a field containing a date in ISO 8601 format which describes the last time that an Elastic Prebuilt Detection rule was created and subsequently updated by the TRaDe team, the team responsible for creating detection rules. Its usage is detailed in this [ticket](https://github.com/elastic/detection-rules/issues/2826).
+
+This field will be optional in both the API schema and the internal rule schema, since this value will not exist for prebuilt rules until a new version of each rule which includes this field in the prebuilt rule asset is published by the TRaDE team, and the user installs it or updates to it.
+
+### Changes needed in rule schema
+
+As detailed further down in the [Plan for carrying out migrations of rule SOs](#plan-for-carrying-out-migrations-of-rule-sos) section, we will be performing incremental migration-on-write on our saved objects to get to our new schemas (see details in linked section).
+
+This means that we need to differentiate between changes to the internal rule schema and the API schema.
+
+#### API schema
+
+##### API request and response rule schema
+
+**In the API schema** the `rule_source` will be a required, as it will allow us to differentiate between Elastic prebuilt rules, and custom rules. 
+
+Notice, as well, that `immutable` and `rule_source` will continue to be part **only of the API response schema**, and never form part of the **request parameters**.
+
+The OpenAPI schema will need to be modified so:
+
+_Source: [x-pack/plugins/security_solution/common/api/detection_engine/model/rule_schema/common_attributes.schema.yaml](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/common/api/detection_engine/model/rule_schema/common_attributes.schema.yaml)_
+
+```yaml
+#  [... file continues above...]
+
+# Add deprecation warning to `immutable` field
+IsRuleImmutable:
+  type: boolean
+  description: '[DEPRECATION WARNING - This field is deprecated and will be removed in a future release. Use the prebuilt field to determine if a rule is an Elastic prebuilt rule or a custom rule.] - Determines whether the rule is immutable (i.e. cannot be modified by the user).'
+
+IsExternalRuleCustomized:
+  type: boolean
+  description: Determines whether an external/prebuilt rule has been customized by the user (i.e. any of its fields have been modified and diverged from the base value).
+
+ExternalSourceUpdatedAt:
+  type: string
+  format: date-time
+  description: The date and time that the external/prebuilt rule was last updated in its source repository.
```

Comment:
Could we please make a note in the RFC that we're not going to add this field to the schema in the initial implementation because of the dependency on TRADE?